app/http.php

```diff
@@ -0,0 +1,1039 @@
+            $wasabiAcl = 'private';
+            return new Wasabi($root, $wasabiAccessKey, $wasabiSecretKey, $wasabiBucket, $wasabiRegion, $wasabiAcl);
+    }
+}
+
+function removeAllRuntimes(Pool $orchestrationPool): void
+{
+    Console::log('Cleaning up containers before shutdown...');
+
+    $connection = $orchestrationPool->pop();
+    $orchestration = $connection->getResource();
+    $functionsToRemove = $orchestration->list(['label' => 'openruntimes-executor=' . System::getHostname()]);
+    $orchestrationPool->push($connection);
+
+    if (\count($functionsToRemove) === 0) {
+        Console::info('No containers found to clean up.');
+    }
+
+    $callables = [];
+
+    foreach ($functionsToRemove as $container) {
+        $callables[] = function () use ($container, $orchestrationPool) {
+            try {
+                $connection = $orchestrationPool->pop();
+                $orchestration = $connection->getResource();
+                $orchestration->remove($container->getId(), true);
+                Console::success('Removed container ' . $container->getName());
+            } catch (\Throwable $th) {
+                Console::error('Failed to remove container: ' . $container->getName());
+                Console::error($th);
+            } finally {
+                isset($connection) && $orchestrationPool->push($connection);
+            }
+        };
+    }
+
+    /** @phpstan-ignore-next-line */
+    Swoole\Coroutine\batch(
+        $callables
+    );
+
+    Console::success('Done.');
+}
+
+App::post('/v1/runtimes')
+    ->desc("Create a new runtime server")
+    ->param('runtimeId', '', new Text(64), 'Unique runtime ID.')
+    ->param('image', '', new Text(128), 'Base image name of the runtime.')
+    ->param('source', '', new Text(0), 'Path to source files.')
+    ->param('entrypoint', '', new Text(256), 'Entrypoint of the code file.')
+    ->param('destination', '', new Text(0), 'Destination folder to store build files into.', true)
+    ->param('variables', [], new Assoc(), 'Environment Variables required for the build.', true)
+    ->param('commands', [], new ArrayList(new Text(1024), 100), 'Commands required to build the container. Maximum of 100 commands are allowed, each 1024 characters long.', true)
+    ->param('workdir', '', new Text(256), 'Working directory.', true)
+    ->param('remove', false, new Boolean(), 'Remove a runtime after execution.', true)
+    ->inject('orchestrationPool')
+    ->inject('activeRuntimes')
+    ->inject('response')
+    ->action(function (string $runtimeId, string $image, string $source, string $entrypoint, string $destination, array $variables, array $commands, string $workdir, bool $remove, Pool $orchestrationPool, Table $activeRuntimes, Response $response) {
+        $activeRuntimeId = $runtimeId; // Used with Swoole table (key)
+        $runtimeId = System::getHostname() . '-' . $runtimeId; // Used in Docker (name)
```

Comment:
Something is not clear to me. We use the runtimeId at start but then we immediately overwrite it, why do we still accept it as user input?

Reply:
Executor needs to do this in order to identify runtimes properly (to allow multiple executors on the same host machine).

Client should not be passing this as he should not even know there is some executor hostname prefix.